### PR TITLE
#191 Check ready state of redis before using it

### PR DESCRIPTION
**Pull request by mcweba**

### CODE REVIEW

File: src/main/java/org/swisspush/redisques/util/ResourcesUtils.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     * Loads the resource with the provided name from the classpath. When param {@code exceptionWhenNotFound}
+     * set to true, a {@link RuntimeException} is thrown when the resource cannot be loaded.
+     * </p>
+     *
+     * @param resourceName the name of the resource to load
+     * @param exceptionWhenNotFound throw a {@link RuntimeException} when the resource could not be loaded
+     * @throws RuntimeException when {@code exceptionWhenNotFound} is set to true and resource cannot be loaded
+     * @return The content of the resource or null
+     */
+    public static String loadResource(String resourceName, boolean exceptionWhenNotFound) {
```
Comment (hiddenalpha) [inline src/main/java/org/swisspush/redisques/util/ResourcesUtils.java:36]:
```patch
-  loadResource
+  loadUtf8ResourceAsString
```

File: src/main/java/org/swisspush/redisques/util/ResourcesUtils.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        try {
+            URL url = Resources.getResource(resourceName);
+            return Resources.toString(url, Charsets.UTF_8);
+        } catch (Exception e) {
+            log.error("Error loading resource '{}'", resourceName, e);
+            if(exceptionWhenNotFound){
+                throw new RuntimeException("Error loading required resource '"+resourceName+"'");
+            }
```
Comment (hiddenalpha) [inline src/main/java/org/swisspush/redisques/util/ResourcesUtils.java:44]:
```patch
-  log.error("Error loading resource '{}'", resourceName, e);
-  if(exceptionWhenNotFound){
-      throw new RuntimeException("Error loading required resource '"+resourceName+"'");
-  }
-  return null;
+  if(exceptionWhenNotFound){
+      throw new RuntimeException("Error loading required resource '"+resourceName+"'", e);
+  }else{
+      log.warn("Error loading resource '{}'", resourceName, e);
+      return null;
+  }
```

**Review by hiddenalpha — COMMENTED** [review]
**TODO**: Somebody of PaISA (which has permission to get the output of "*our*" redis INFO command) should verify that the output "*our*" redis version produces works with this new feature.

File: src/main/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandler.java [hunk 3]
```diff
@@ -271,6 +271,12 @@ private void respondMethodNotAllowed(RoutingContext ctx) {
         respondWith(StatusCode.METHOD_NOT_ALLOWED, ctx.request());
     }
 
+    private void toggleReady(RoutingContext ctx) {
+        log.error("xxx toggle ready");
```
[the remark this answers is not in the document]
Comment (mcweba) [reply src/main/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandler.java:275]: No it's not an error and I should have removed this method. It was only used during development

**Pull request merged**